main

```diff
@@ -0,0 +1,158 @@
+require './app'
+
+class Main
+  def initialize
+    @app = App.new
+    @option = ''
+    @type = ''
+    @age = ''
+    @name = ''
+    @permission = ''
+    @specialization = ''
+    @title = ''
+    @author = ''
+    @book = ''
+    @person = ''
+    @date = ''
+    @id = ''
+  end
+
+  def menu_content
+    puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def check_input(str)
+    loop do
+      print str
+      @option = gets.chomp.strip
+      break if yield
+    end
+    @option
+  end
+
+  def menu
+    menu_content
+    @option = check_input('') { %w[1 2 3 4 5 6 7].include?(@option) }
+  end
+
+  def list_all_books
+    puts 'The book list is empty!' if @app.list_books.length.zero?
+    @app.list_books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'The people list is empty!' if @app.list_people.length.zero?
+    @app.list_people.each do |people|
+      puts "[#{people.class}] Name: #{people.name}, ID: #{people.id}, Age: #{people.age}"
+    end
+  end
+
+  def create_a_person
+    @type = check_input('Do you want to create a student (1) or a teacher (2)? [Input the number]: ') do
+      %w[1 2].include?(@option)
+    end
+    @age = check_input('Age: ') { @option.match?(/^\d+$/) }
+    @name = check_input('Name: ') { @option != '' }
+    if @type == '1'
+      @permission = check_input('Has parent permission? [Y/N]: ') { %w[y n].include?(@option.downcase) }
+      @app.create_new_student(@age, @name, @permission)
+    else
+      @specialization = check_input('Specialization: ') { @option != '' }
+      @app.create_new_teacher(@specialization, @age, @name)
+    end
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] I think you have an error in this method so when I create a new person it shows me this error in the terminal 
so kindly work on it to fix the problem 🙏🏻💯🚀🚀
![image](https://user-images.githubusercontent.com/58553711/187808730-97424cad-89b2-4d07-8f2c-ef098ae219fe.png)